lib/base-compiler.js

```diff
@@ -172,12 +179,27 @@ Compile.prototype.compile = function (source, options, filters) {
                         result.asm = "<Compilation failed>";
                         return result;
                     }
+                    result.hasOptOutput = false;
+                    if(self.couldHaveOptOutput(options, self.compiler.version)) {
+                        const optPath = path.join(dirPath, outputFilebase + ".opt.yaml");
+                        if(fs.existsSync(optPath)) {
```

Comment:
(i thought i had responded to this)
There is no async equivalent of this they depreciated in favour of a stat call, which is a little uglier and harder to read IMO.  I can't see this one blocking too long.